debezium-server/debezium-server-infinispan/src/main/java/io/debezium/server/infinispan/InfinispanSinkConsumer.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.server.infinispan;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import javax.enterprise.context.Dependent;
+import javax.enterprise.inject.Instance;
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.client.hotrod.RemoteCacheManager;
+import org.infinispan.client.hotrod.configuration.ConfigurationBuilder;
+import org.infinispan.client.hotrod.impl.ConfigurationProperties;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.DebeziumException;
+import io.debezium.engine.ChangeEvent;
+import io.debezium.engine.DebeziumEngine;
+import io.debezium.server.BaseChangeConsumer;
+import io.debezium.server.CustomConsumerBuilder;
+
+/**
+ * An implementation of the {@link DebeziumEngine.ChangeConsumer} interface that publishes change event messages to predefined Infinispan cache.
+ *
+ * @author vjuranek
+ */
+@Named("infinispan")
+@Dependent
+public class InfinispanSinkConsumer extends BaseChangeConsumer implements DebeziumEngine.ChangeConsumer<ChangeEvent<Object, Object>> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(InfinispanSinkConsumer.class);
+
+    private static final String CONF_PREFIX = "debezium.sink.infinispan.";
+    private static final String SERVER_HOST = CONF_PREFIX + "server.host";
+    private static final String SERVER_PORT = CONF_PREFIX + "server.port";
+    private static final String CACHE_NAME = CONF_PREFIX + "cache";
+    private static final String USER_NAME = CONF_PREFIX + "user";
+    private static final String PASSWORD = CONF_PREFIX + "password";
+
+    private RemoteCacheManager remoteCacheManager;
+    private RemoteCache cache;
+
+    @Inject
+    @CustomConsumerBuilder
+    Instance<RemoteCache> customCache;
+
+    @PostConstruct
+    void connect() {
+        if (customCache.isResolvable()) {
+            cache = customCache.get();
+            LOGGER.info("Obtained custom cache with configuration '{}'", cache.getRemoteCacheContainer().getConfiguration());
+            return;
+        }
+
+        final Config config = ConfigProvider.getConfig();
+        final String serverHost = config.getValue(SERVER_HOST, String.class);
+        final String cacheName = config.getValue(CACHE_NAME, String.class);
```

Comment:
cache name is mandatory but what if it is made optional and when not provided then topic name is used as the cache name so you can feed multiple caches? Alo in that case the custom stream name mapper can/should be used as well.

Reply:
Most of the other sink have a notion of stream or topic where which you can specify along with the records and it's handled on the server. This is not the case of Infinispan - you have to select the cache where you want to send the records and there is no possibility to just send the the records along with the cache names and tell Infinispan to sort it out before storing. If we want to support it, we would need to create multiple `RemoteCache`s, define the mapping and sort and send the records to respective cache ourselves. I'd say more typical Infinispan usage is to use one/few big caches rather than many small caches, so I'd support only one cache for now. We can extend it in the future if there will be any demand for it.